docs/source/common/lightning_cli.rst

```diff
@@ -665,69 +665,131 @@ Optimizers and learning rate schedulers
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Optimizers and learning rate schedulers can also be made configurable. The most common case is when a model only has a
-single optimizer and optionally a single learning rate scheduler. In this case the model's
-:class:`~pytorch_lightning.core.lightning.LightningModule` could be left without implementing the
-:code:`configure_optimizers` method since it is normally always the same and just adds boilerplate. The following code
-snippet shows how to implement it:
+single optimizer and optionally a single learning rate scheduler. In this case, the model's
+:meth:`~pytorch_lightning.core.lightning.LightningModule.configure_optimizers` could be left unimplemented since it is
+normally always the same and just adds boilerplate.
 
-.. testcode::
-
-    import torch
-
-
-    class MyLightningCLI(LightningCLI):
-        def add_arguments_to_parser(self, parser):
-            parser.add_optimizer_args(torch.optim.Adam)
-            parser.add_lr_scheduler_args(torch.optim.lr_scheduler.ExponentialLR)
+The CLI works out-of-the-box with PyTorch's built-in optimizers and learning rate schedulers when
+at most one of each is used.
+Only the optimizer or scheduler name needs to be passed, optionally with its ``__init__`` arguments:
 
+.. code-block:: bash
 
-    cli = MyLightningCLI(MyModel)
+    $ python trainer.py fit --optimizer=Adam --optimizer.lr=0.01 --lr_scheduler=ExponentialLR --lr_scheduler.gamma=0.1
 
-With this the :code:`configure_optimizers` method is automatically implemented and in the config the :code:`optimizer`
-and :code:`lr_scheduler` groups would accept all of the options for the given classes, in this example :code:`Adam` and
-:code:`ExponentialLR`. Therefore, the config file would be structured like:
+A corresponding example of the config file would be:
 
 .. code-block:: yaml
 
     optimizer:
-      lr: 0.01
+      class_path: torch.optim.Adam
+      init_args:
+        lr: 0.01
     lr_scheduler:
-      gamma: 0.2
+      class_path: torch.optim.lr_scheduler.ExponentialLR
+      init_args:
+        gamma: 0.1
     model:
       ...
     trainer:
       ...
 
-And any of these arguments could be passed directly through command line. For example:
+.. note::
+
+    This short-hand notation is only supported in the shell and not inside a configuration file. The configuration file
+    generated by calling the previous command with ``--print_config`` will have the ``class_path`` notation.
+
+Furthermore, you can register your own optimizers and/or learning rate schedulers as follows:
+
+.. code-block:: python
+
+    from pytorch_lightning.utilities.cli import OPTIMIZER_REGISTRY, LR_SCHEDULER_REGISTRY
+
+
+    @OPTIMIZER_REGISTRY
+    class CustomAdam(torch.optim.Adam):
+        ...
+
+
+    @LR_SCHEDULER_REGISTRY
+    class CustomCosineAnnealingLR(torch.optim.lr_scheduler.CosineAnnealingLR):
```

Comment:
can you give a recommendation what to do if I want to register a optimizer or scheduler from another library?

Reply:
`OPTIMIZER_REGISTRY.register_package(torch.optim, Optimizer)` ?

Reply:
That's right, sir.

Added to the docs